packages/client-sdk-nodejs/src/internal/data-client.ts

```diff
@@ -2226,9 +2233,12 @@ export class DataClient {
           metadata,
           {interceptors: this.interceptors},
           (err, resp) => {
-            if (resp?.missing) {
+            if (
+              resp?.missing ||
+              resp?.element_rank?.result === ECacheResult.Miss
+            ) {
               resolve(new CacheSortedSetGetRank.Miss());
-            } else if (resp?.element_rank) {
+            } else if (resp?.element_rank?.result === ECacheResult.Hit) {
               if (resp?.element_rank.rank === undefined) {
```

Comment:
Late thought - `if (resp?.element_rank.rank === undefined)` can never be true with the new protogen ts files if it is a hit, do we still need this check?